mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala

```diff
@@ -438,6 +438,10 @@ private[ml] object SummaryBuilderImpl extends Logging {
         while (i < len) {
           realVariance(i) = (currM2n(i) + deltaMean(i) * deltaMean(i) * weightSum(i) *
             (totalWeightSum - weightSum(i)) / totalWeightSum) / denominator
+          // Because of numerical error, it is possible to get negative real variance
+          if (realVariance(i) < 0.0) {
```

Comment:
Just use `math.max(0.0 ...)` in the line above? no need to assign it twice.

Reply:
Hmm.. `WeightedLeastSquares` use another way to compute variance `Var(X) = E(X^2) - E(X)^2`. But it seems also possible to have this problem.